### PR TITLE
SDL Multitouch!

### PR DESCRIPTION
**Pull request by TheBrokenRail**
This adds multitouch support to the SDL build.

### CODE REVIEW

File: platforms/sdl/main.cpp [hunk 1]
```diff
@@ -54,6 +55,60 @@ static int TranslateSDLKeyCodeToVirtual(int sdlCode)
 	return SDLVK_UNKNOWN;
 }
 
+// Touch
+#define TOUCH_IDS_SIZE (MAX_TOUCHES - 1) // ID 0 Is Reserved For The Mouse
```
Comment (iProgramMC) [inline platforms/sdl/main.cpp:59]: `Multitouch` does not reserve IDs for the mouse. Also the `Mouse` and the `Multitouch` are checked separately, have separate event queues, and are basically entirely separately, except for the fact that Multitouch is implemented on top of an array of `MouseDevice`s.  So I don't think you need this.

File: source/CMakeLists.txt [hunk 2]
```diff
@@ -90,22 +90,22 @@ add_library(reminecraftpe-core STATIC
...
-	client/player/input/CustomInputHolder.cpp
-	client/player/input/IInputHolder.cpp
-	client/player/input/IMoveInput.cpp
-	client/player/input/ITouchScreenModel.cpp
-	client/player/input/TouchAreaModel.cpp
-	client/player/input/TouchInputHolder.cpp
-	client/player/input/TouchscreenInput_TestFps.cpp
-	client/player/input/UnifiedTurnBuild.cpp
     client/network/ClientSideNetworkHandler.cpp
+    client/player/input/IBuildInput.cpp
```
Comment (iProgramMC) [inline source/CMakeLists.txt:93]: Why were these moved to insert `ClientSideNetworkHandler.cpp` in the middle of the `client/player/input` group?

**Comment by iProgramMC** [comment]
Other than these, and the fact that there's more changes than necessary due to you changing the indent characters, your PR looks fine to me.

File: source/CMakeLists.txt (hunk 2 above)
Comment (TheBrokenRail) [reply source/CMakeLists.txt:93]: Because I started this PR before you fixed the CMake and the rebase messed things up. I'll fix it.

File: platforms/sdl/main.cpp (hunk 1 above)
Comment (TheBrokenRail) [reply platforms/sdl/main.cpp:59]: If mouse events don't also send multitouch events, things become fairly buggy. But for mouse event callbacks to send multitouch events, they need a touch ID. So I reserved ID 0.

**Comment by TheBrokenRail** [comment]
Sorry about the indents, I accidentally used 4-space indents when starting this PR, and fixing that seems to have fixed other code as well.

File: source/CMakeLists.txt (hunk 2 above)
Comment (TheBrokenRail) [reply source/CMakeLists.txt:93]: This is fixed now.
